Failure to format code with a custom annotation with Scala 3

- **Version**: 2.4.3
- **Integration**: sbt-scalafmt
- **Configuration**:
```
version = "3.0.6"
runner.dialect = scala3
```

## Steps

Given code like this:
```scala
import caliban.schema.Annotations.GQLInterface

@GQLInterface
enum MyADT2 {
  case A(a: Int)
  case B(a: Int)
}
```

When I run scalafmt like this:

```bash
sbt scalafmt
```

## Problem

Scalafmt fails with this error
```scala
[error] /Users/pierre/GIT/caliban/core/src/test/scala-3/caliban/Scala3SpecificSpec.scala:20: error: [dialect scala3]  Invalid modifier @GQLInterface for: enum MyADT2
[error]   @GQLInterface
[error]   ^: /Users/pierre/GIT/caliban/core/src/test/scala-3/caliban/Scala3SpecificSpec.scala
[error] (core / Test / scalafmt) /Users/pierre/GIT/caliban/core/src/test/scala-3/caliban/Scala3SpecificSpec.scala:20: error: [dialect scala3]  Invalid modifier @GQLInterface for: enum MyADT2
[error]   @GQLInterface
[error]   ^: /Users/pierre/GIT/caliban/core/src/test/scala-3/caliban/Scala3SpecificSpec.scala
```

## Expectation

No error.

--- a/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala
+++ b/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala
@@ -989,16 +989,20 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
 
   def onlyAllowedMods[M1 <: Mod, M2 <: Mod](mods: List[Mod], culprit: String)(
       implicit classifier1: Classifier[Mod, M1],
-      tag1: ClassTag[M1],
-      classifier2: Classifier[Mod, M2],
-      tag2: ClassTag[M2]
-  ) = {
+      classifier2: Classifier[Mod, M2]
+  ): Unit = {
+    onlyAllowedMods(mods, List(classifier1.apply, classifier2.apply), culprit)
+  }
+
+  def onlyAllowedMods(mods: List[Mod], matchers: List[Mod => Boolean], culprit: String): Unit = {
     mods
       .foreach {
-        case m if classifier1.apply(m) || classifier2.apply(m) =>
+        case m if matchers.exists(f => f(m)) =>
         case m => syntaxError(s" Invalid modifier ${m}${formatCulprit(culprit)}", at = m)
       }
   }
+
+  def summonClassifierFunc[A, B](implicit v: Classifier[A, B]): A => Boolean = v.apply(_)
 
   def onlyAcceptMod[M <: Mod: ClassTag, T <: Token: TokenInfo](
       mods: List[Mod],
@@ -4627,7 +4631,16 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
 
     val enumName = typeName()
     val culprit = s"enum $enumName"
-    onlyAllowedMods[Mod.Private, Mod.Protected](mods, culprit)
+
+    onlyAllowedMods(
+      mods,
+      List(
+        summonClassifierFunc[Mod, Mod.Private],
+        summonClassifierFunc[Mod, Mod.Protected],
+        summonClassifierFunc[Mod, Mod.Annot]
+      ),
+      culprit
+    )
 
     val typeParams = typeParamClauseOpt(
       ownerIsType = true,

--- a/tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EnumSuite.scala
+++ b/tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EnumSuite.scala
@@ -385,6 +385,30 @@ class EnumSuite extends BaseDottySuite {
     )
   }
 
+  test("enum-annotated") {
+    val code =
+      """|@annot
+         |enum A:
+         |  case B, C
+      """.stripMargin
+    val expected = "@annot enum A { case B, C }"
+    runTestAssert[Stat](code, assertLayout = Some(expected))(
+      Defn.Enum(
+        List(Mod.Annot(Init(Type.Name("annot"), Name(""), Nil))),
+        Type.Name("A"),
+        Nil,
+        Ctor.Primary(Nil, Name(""), Nil),
+        Template(
+          Nil,
+          Nil,
+          Self(Name(""), None),
+          List(Defn.RepeatedEnumCase(Nil, List(Term.Name("B"), Term.Name("C")))),
+          Nil
+        )
+      )
+    )
+  }
+
   test("enum-self-braces") {
     val code =
       """|enum A { self =>

